packages/vats/src/ibc.js

```diff
@@ -661,26 +682,57 @@ export const prepareIBCProtocol = (zone, powers) => {
             source_port: portID,
           } = fullPacket;
 
-          /** @type {VowKit<Bytes>} */
-          const { vow } = util.findAckKit(channelID, portID, sequence);
-          return vow;
+          const vow = util.extractAckKit(
+            channelID,
+            portID,
+            BigInt(sequence),
+            'vow',
+          );
+          return NonNullish(vow);
         },
 
         /**
+         * @template {'resolver' | 'vow'} K
          * @param {IBCChannelID} channelID
          * @param {IBCPortID} portID
          * @param {bigint} sequence
+         * @param {K} key
+         * @returns {VowKit<Bytes>[K] | undefined}
          */
-        findAckKit(channelID, portID, sequence) {
+        extractAckKit(channelID, portID, sequence, key) {
           const { channelKeyToSeqAck } = this.state;
           const channelKey = `${channelID}:${portID}`;
           const seqToAck = channelKeyToSeqAck.get(channelKey);
-          if (seqToAck.has(sequence)) {
-            return seqToAck.get(sequence);
+
+          // Try also looking up the sequence as a number in case it was added
+          // to the map before the upgrade to bigints.
+          for (const seq of [sequence, Number(sequence)]) {
+            if (!seqToAck.has(seq)) {
+              continue;
+            }
+
+            const { [key]: extracted, ...rest } = seqToAck.get(seq);
```

Comment:
Suggestion to minimize store calls:
```suggestion
            const found = seqToAck.get(seq);
            if (!found) continue;

            const { [key]: extracted, ...rest } = found;
```

Reply:
`seqToAck` is a `MapStore`, so `seqToAck.get(seq)` throws if `!seqToAck.has(seq)`.  I'd prefer not to introduce a `try-catch` block.

Reply:
`seqToAck` is a `MapStore`, so `seqToAck.get(seq)` throws if `!seqToAck.has(seq)`.  I'd prefer not to introduce a `try-catch` block.  Would you still recommend minimizing store calls?

Reply:
No, I forgot about that difference between Map and MapStore.